glue/plugins/wcs_autolinking/wcs_autolinking.py

```diff
@@ -60,43 +86,144 @@ def __init__(self, data1=None, data2=None, cids1=None, cids2=None):
+            wcs2_celestial_axis_physical_types = None
 
             if not wcs1.has_celestial or not wcs2.has_celestial:
-                raise IncompatibleWCS("Can't create WCS link between {0} and {1}".format(data1.label, data2.label))
+                slicing_axes_celestial1 = []
+                slicing_axes_celestial2 = []
+            elif wcs1.has_celestial and wcs2.has_celestial:
+                wcs1_celestial_axis_physical_types = wcs1.celestial.world_axis_physical_types
+                wcs2_celestial_axis_physical_types = wcs2.celestial.world_axis_physical_types
+
+                cids1 = data1.pixel_component_ids
+                cids2 = data2.pixel_component_ids
+                cids1_celestial = [cids1[wcs1.wcs.naxis - wcs1.wcs.lng - 1],
+                                   cids1[wcs1.wcs.naxis - wcs1.wcs.lat - 1]]
+                cids2_celestial = [cids2[wcs2.wcs.naxis - wcs2.wcs.lng - 1],
+                                   cids2[wcs2.wcs.naxis - wcs2.wcs.lat - 1]]
+
+                slicing_axes_celestial1 = [cids1_celestial[0].axis, cids1_celestial[1].axis]
+                slicing_axes_celestial1 = sorted(slicing_axes_celestial1, key=str, reverse=False)
+
+                slicing_axes_celestial2 = [cids2_celestial[0].axis, cids2_celestial[1].axis]
+                slicing_axes_celestial2 = sorted(slicing_axes_celestial2, key=str, reverse=False)
+            elif wcs1.has_celestial and not wcs2.has_celestial:
+                wcs1_celestial_axis_physical_types = wcs1.celestial.world_axis_physical_types
+                cids1 = data1.pixel_component_ids
+                cids1_celestial = [cids1[wcs1.wcs.naxis - wcs1.wcs.lng - 1],
+                                   cids1[wcs1.wcs.naxis - wcs1.wcs.lat - 1]]
+                slicing_axes_celestial1 = [cids1_celestial[0].axis, cids1_celestial[1].axis]
+                slicing_axes_celestial1 = sorted(slicing_axes_celestial1, key=str, reverse=False)
+
+                slicing_axes_celestial2 = []
+
+            elif not wcs1.has_celestial and wcs2.has_celestial:
+                slicing_axes_celestial1 = []
+
+                wcs2_celestial_axis_physical_types = wcs2.celestial.world_axis_physical_types
+                cids2 = data2.pixel_component_ids
+                cids2_celestial = [cids2[wcs2.wcs.naxis - wcs2.wcs.lng - 1],
+                                   cids2[wcs2.wcs.naxis - wcs2.wcs.lat - 1]]
+                slicing_axes_celestial2 = [cids2_celestial[0].axis, cids2_celestial[1].axis]
+                slicing_axes_celestial2 = sorted(slicing_axes_celestial2, key=str, reverse=False)
+
+            # Collect all apparently matching axes in two lists for the two wcs objects being linked up
+            if wcs1_celestial_axis_physical_types is not None:
+                wcs1_sliced_physical_types = wcs1_celestial_axis_physical_types
+            else:
+                wcs1_sliced_physical_types = []
+            slicing_axes1 = slicing_axes_celestial1
```

Comment:
Maybe you can simply use ``slicing_axes1`` instead of ``slicing_axes_celestial1`` in the if statement above?

Reply:
Sure, guess I was trying to separate the celestial axes from the rest originally...